noir-r1cs/src/noir_witness.rs

```diff
@@ -22,6 +22,7 @@ pub struct NoirWitnessGenerator {
     // Note: Abi uses an [internally tagged] enum format in Serde, which is not compatible
     // with some schemaless formats like Postcard.
     // [internally-tagged]: https://serde.rs/enum-representations.html
+    // TODO: serializes the ABI as a json string. Something like CBOR might be better.
```

Comment:
Agree. Now I remember it all again. This is a pretty ugly hack. Fortunately the Abi object is small so it has little impact.

Reply:
Yeah, I'm leaving the TODO - I don't think it's worth sinking time into it now